### PR TITLE
Joe/french

### PR DESCRIPTION
**Pull request by joe-lazzarini**
### Add french support in scripts

Partial solution for #168 
Parse French words/phrases and add to DB
Parse French phrases in txt and replace spaces with &fr

### Related Github Issues

- #[Number of Issue]
- Or, remember to link to an issue on the sidebar

### Anything Else

- Related PRs
- Bugs
- Concerns

### CODE REVIEW

**Comment by joe-lazzarini** [comment]
@Fattimo please verify that &fr is formatted as you desire (see 3rd line on page 1)

File: public/book/pages/1/1.txt [hunk 1]
```diff
@@ -1,5 +1,5 @@
 A green and yellow parrot, which hung in a cage outside the door, kept repeating over and over:
-“Allez vous-en! Allez vous-en! Sapristi! That’s all right!”
+“Allez&frvous-en!&frAllez&frvous-en!&frSapristi! That’s all right!”
```
Comment (Fattimo) [inline public/book/pages/1/1.txt:2]: Be careful with preceding spaces. I would expect this line to look like:

`“Allez&frvous-en! Allez&frvous-en! Sapristi! That’s all right!”`

File: public/book/pages/13/4.txt [hunk 2]
```diff
@@ -1,5 +1,5 @@
 &ni;  sloping keel of the overturned boat. He was reading from a book. Tonie was no longer with him. She wondered what had become of the rest of the party. She peeped out at him two or three times as she stood washing herself in the little basin between the windows.
-Madame Antoine had laid some coarse, clean towels upon a chair, and had placed a box of poudre de riz within easy reach. Edna dabbed the powder upon her nose and cheeks as she looked at herself closely in the little distorted mirror which hung on the wall above the basin. Her eyes were bright and wide awake and her face glowed.
+Madame Antoine had laid some coarse, clean towels upon a chair, and had placed a box of poudre&frde&frriz within easy reach. Edna dabbed the powder upon her nose and cheeks as she looked at herself closely in the little distorted mirror which hung on the wall above the basin. Her eyes were bright and wide awake and her face glowed.
```
Comment (Fattimo) [inline public/book/pages/13/4.txt:2]: This one is good 👍

File: public/book/pages/14/2.txt [hunk 3]
```diff
@@ -4,4 +4,4 @@
...
 She wondered why Robert had gone away and left her. It did not occur to her to think he might have grown tired of being with her the livelong day. She was not tired, and she felt that he was not. She regretted that he had gone. It was so much more natural to have him stay when he was not absolutely required to leave her.
-As Edna waited for her husband she sang low a little song that Robert had sung as they crossed the bay. It began with “Ah! si tu savais,” and every verse ended with “si tu savais.”
+As Edna waited for her husband she sang low a little song that Robert had sung as they crossed the bay. It began with “Ah!&frsi&frtu&frsavais,” and every verse ended with “si tu savais.”
```
Comment (Fattimo) [inline public/book/pages/14/2.txt:7]: no &fr preceding `si`

File: public/book/pages/30/5.txt [hunk 4]
```diff
@@ -1,6 +1,6 @@
...
+The moments glided on, while a feeling of good fellowship passed around the circle like a mystic cord, holding and binding these people together with jest and laughter. Monsieur Ratignolle was the first to break the pleasant charm. At ten o’clock he excused himself. Madame Ratignolle was waiting for him at home. She was bien&frsouffrante, and she was filled with vague dread, which only her husband’s presence could allay.
+Mademoiselle Reisz arose with Monsieur Ratignolle, who offered to escort her to the car. She had eaten well; she had tasted the good, rich wines, and they must have turned her head, for she bowed pleasantly to all as she withdrew from table. She kissed Edna upon the shoulder, and whispered: “Bonne&frnuit,&frma&frreine;&frsoyez&frsage.” She had been a little bewildered upon rising, or rather, descending from her cushions, and Monsieur Ratignolle gallantly took her arm and led her away.
```
Comment (Fattimo) [inline public/book/pages/30/5.txt:4]: if the second one is all one phrase, then this is ok

File: public/book/pages/5/1.txt [hunk 5]
```diff
@@ -3,5 +3,5 @@ He had lived in her shadow during the past month. No one thought anything of it.
...
 Mrs. Pontellier liked to sit and gaze at her fair companion as she might look upon a faultless Madonna.
 “Could any one fathom the cruelty beneath that fair exterior?” murmured Robert. “She knew that I adored her once, and she let me adore her. It was ‘Robert, come; go; stand up; sit down; do this; do that; see if the baby sleeps; my thimble, please, that I left God knows where. Come and read Daudet to me while I sew.’”
-“Par exemple! I never had to ask. You were always there under my feet, like a troublesome cat.”
-“You mean like an adoring dog. And just as soon as Ratignolle appeared on the scene, then it was like a dog. ‘Passez! Adieu! Allez vous-en!’”
+“Par&frexemple! I never had to ask. You were always there under my feet, like a troublesome cat.”
+“You mean like an adoring dog. And just as soon as Ratignolle appeared on the scene, then it was like a dog. ‘Passez!&frAdieu!&frAllez&frvous-en!’”
```
Comment (Fattimo) [inline public/book/pages/5/1.txt:7]: if all the same phrase, then ok

**Review by Fattimo — COMMENTED** [review]
Overall should be ok. Believe there is on recurring issue with "...! si tu savais", as it is adding a delimiter between the and the french phrase.

File: public/book/pages/1/1.txt (hunk 1 above)
Comment (joe-lazzarini) [reply public/book/pages/1/1.txt:2]: I'm not sure if it makes a difference, but this is actually one phrase. Some of them have weird formats

french

**Review by cmurray191 — APPROVED** [review]
I looked through the pages and it seems good to me

**Comment by Fattimo** [comment]
@joe-lazzarini have phrases been parsed into the db yet? I only see definitions for individual words

**Comment by Fattimo** [comment]
also, when you do the parsing for phrases, please omit any punctuation (space separated words are ok)

**Comment by Fattimo** [comment]
and all lowercase

**Comment by Fattimo** [comment]
> @joe-lazzarini have phrases been parsed into the db yet? I only see definitions for individual words

nevermind, found them

**Comment by joe-lazzarini** [comment]
Already was lowercase, joe/french updated removing punctuation. Use it or not, it was only 3 lines of code

**Pull request merged**